apps/individuals/api/serializers.py

```diff
@@ -24,33 +30,198 @@ class Meta:
 
-class IndividualSerializer(serializers.ModelSerializer):
-    employment_details = EmploymentDetailSerializer(
-        many=True,
-        read_only=True
-    )
-    next_of_kin = NextOfKinSerializer(
-        many=True,
-        read_only=True
-    )
-    gender_display = serializers.CharField(
-        source='get_gender_display',
-        read_only=True
-    )
-    identification_type_display = serializers.CharField(
-        source='get_identification_type_display',
-        read_only=True
+class ContactDetailsSerializer(serializers.ModelSerializer):
+    mobile_phone = serializers.ListField(
+        child=serializers.CharField(max_length=15),
+        allow_empty=False
     )
+    class Meta:
+        model = IndividualContactDetail
+        fields = ['id','individual_id' ,'mobile_phone', 'email',]
+    
+class IndividualSerializer(serializers.ModelSerializer):
+    employment_details = EmploymentDetailSerializer(many=True,required=False)
+    next_of_kin = NextOfKinSerializer(many=True, required=False)
+    gender_display = serializers.CharField(source='get_gender_display',read_only=True)
+    identification_type_display = serializers.CharField(source='get_identification_type_display',read_only=True)
+    addresses = AddressSerializer(many=True,required=True)
+    notes = NoteSerializer(many=True, required=False)
+    documents = DocumentSerializer(many=True, required=False)
+    contact_details = ContactDetailsSerializer(many=True, required=False)
     
     class Meta:
         model = Individual
         fields = [
             'id', 'first_name', 'last_name', 'full_name',
             'date_of_birth', 'gender', 'gender_display',
             'identification_type', 'identification_type_display',
-            'identification_number', 'email', 'mobile_phone',
-            'landline_phone', 'is_verified', 'is_active',
-            'employment_details', 'next_of_kin',
-            'date_created', 'date_updated'
+            'identification_number','contact_details', 'is_verified', 'is_active',
+            'employment_details', 'next_of_kin', 'documents', 
+            'addresses', 'notes','date_created', 'date_updated'
         ]
-        read_only_fields = ['date_created', 'date_updated']
\ No newline at end of file
+        read_only_fields = ['date_created', 'date_updated']
+
+class IndividualMinimalSerializer(serializers.ModelSerializer):
+    address = serializers.SerializerMethodField()
+    current_employment = serializers.SerializerMethodField()
+    contact_details = ContactDetailsSerializer(many=True, required=False)
+    
+    class Meta:
+        model = Individual
+        fields = ['id', 'first_name', 'last_name', 'identification_number',
+                  'gender','date_of_birth','marital_status','address', 
+                  'current_employment', 'contact_details']
+    
+    # Get the primary address    
+    def get_address(self, obj):
+        primary_address = obj.addresses.filter(is_primary=True).first()
```

Comment:
normalize writing short and straight forward code, normalize using named expressions for cleaner code.

```suggestion
if primary_address := obj.addresses.filter(is_primary=True).first():
    return AddressSerializer(primary_address).data
```